### PR TITLE
Add EbookMaker to run from HTML menu

### PR DESCRIPTION
**Pull request by windymilla**
- Add EBookMaker to Set File Paths dialog.
- Initialise location of EBookMaker.
- Commonise executable location code.
- Add menu entry for EBookMaker.

### CODE REVIEW

**Comment by windymilla** [comment]
@srjfoo wanted to check the running of EBM from within GG on a Mac

**Comment by windymilla** [comment]
New commit adds tidy & kindlegen to path. Local variable means that once sub ebookmaker exits, path will restore to previous value. Tested on Windows by removing path change from run_guiguts.bat, then realised there was no point in putting it back in.
Hopefully ready to merge.

File: src/lib/Guiguts/Utilities.pm [hunk 1]
```diff
@@ -1747,6 +1755,40 @@ sub seeindex {
...
+	my $kindlegenpath = ::catdir( $::lglobal{guigutsdirectory}, 'tools', 'kindlegen' );
+	# local variable means global value will be restored at end of sub
+	local $ENV{PATH} = pathcatdir( pathcatdir( $ENV{PATH}, $tidypath ), $kindlegenpath );
+
+	::runner( $::ebookmakercommand, "--verbose", "--max-depth=3", "--make=epub.images", "--make=kindle.images",
+	          "--output-dir=$outputdir.", "--title=$fname", "$filepath" );
+}
+
+# Concatenate given directories using OS-specific path separator
+sub pathcatdir {
```
Comment (cpeel) [inline src/lib/Guiguts/Utilities.pm:1788]: Can you please add this to the export list so I can use it in `Preferences.pm` as well?

**Comment by cpeel** [comment]
Functionally this worked great for me on Mac. A small request and I think this is ready to merge.

File: src/lib/Guiguts/Utilities.pm (hunk 1 above)
Comment (cpeel) [reply src/lib/Guiguts/Utilities.pm:1788]: Actually, strike that - the code I have in place for `setdefaultpath` doesn't need updating so this is good as-is.

**Comment by srjfoo** [comment]
I realize that I'm late with responding, but thought I'd mention it, anyway -- finally got the right code. ebookmaker worked fine for me, creating epubs, though kindlegen did not. The error message I got for kindlegen:

```
ERROR:root:KindleWriter: kindlegen [Errno 86] Bad CPU type in executable: 'kindlegen'
WARNING:root:
INFO:root: kindle.images made in 0:00:00.005271
```

(And of course, in spite of what it said, it did not make the kindle.images file)

**Comment by windymilla** [comment]
I think I may have seen a comment somewhere that Kindlegen as released
won't run on 64 bit Macs.

Yes. Googling comes up with some hits. One said that it's included in
Kindle Previewer 3, and said the file is here:

/Applications/Kindle\ Previewer\ 3.app/Contents/MacOS/lib/fc/bin/kindlegen

Not sure what we do about that, if that is the problem you have?

**Comment by srjfoo** [comment]
Casey walked me through pulling the 64 bit version kindlegen out of Kindle Previewer 3, so that's what should be in my tools directory. I can check tomorrow to see if that's the case, though -- it's probably time for me to stop for the night.

I know that it works on my system, though, because the last itme I ran ebookmaker from the command line, the mobi was generated quite nicely.

**Comment by cpeel** [comment]
Nigel is correct, the version of kindlegen it is running is the i386 version which won't run on Catalina.

Expecting this to come up in general, the [tools/kindlegen/README.md](https://github.com/DistributedProofreaders/guiguts/tree/master/tools/kindlegen) that we include with the releases talks about this.

**Comment by srjfoo** [comment]
I probably just missed something, then -- I had already extracted the 64 bit kindlegen from KP3. I'll poke some more and make sure it wound up where it was suppsed to be.

**Comment by windymilla** [comment]
Ah. The readme must be where I recalled seeing it. For it to work from the menu entry, kindlegen would have to be in tools/kindlegen below your main GG folder

**Comment by srjfoo** [comment]
Yes, that was it -- I hadn't copied the correct kindlegen into the alpha4 tools folder, so when I copied the alpha4 tools to the working draft, I had the wrong kindlegen. No problem creating the mobi this time. Thanks!

**Pull request merged**